sourcefiles/randomizer.py

```diff
@@ -571,8 +573,11 @@ def get_generated_rom(self) -> bytearray:
 
         return self.out_rom.rom_data.getvalue()
 
-    def write_spoiler_log(self, filename):
-        with open(filename, 'w') as outfile:
+    def write_spoiler_log(self, outfile):
+        if isinstance(outfile, str):
+            with open(outfile, 'w') as real_outfile:
+                self.write_spoiler_log(real_outfile)
```

Comment:
I figure this is a sensible enough way to allow calling this with either a string (filename) or a file-like object.